addons/crm_iap_lead/models/crm_iap_lead_mining_request.py

```diff
@@ -242,22 +242,58 @@ def action_submit(self):
 
     def action_get_lead_action(self):
         self.ensure_one()
+        try_again_action_id = self.env.ref('crm_iap_lead.action_crm_iap_lead_mining_request_try_again').id
         action = self.env["ir.actions.actions"]._for_xml_id("crm.crm_lead_all_leads")
         action['domain'] = [('id', 'in', self.lead_ids.ids), ('type', '=', 'lead')]
         action['help'] = _("""<p class="o_view_nocontent_empty_folder">
             No leads found
         </p><p>
-            No leads could be generated according to your search criteria
-        </p>""")
+            Your search criteria did not return any leads. No credits have been used.
+        </p>
+        <a type="action" name="%s" tabindex="-1">Try Again</a>
+         """, try_again_action_id)
         return action
 
+    def _action_try_again(self):
+        last_mining_request = self.env['crm.iap.lead.mining.request'].search([('user_id', '=', self.env.uid)], order='create_date desc', limit=1)
+        return {
+            'name': _('Generate Leads'),
+            'res_model': 'crm.iap.lead.mining.request',
+            'views': [[False, 'form']],
+            'target': 'new',
+            'type': 'ir.actions.act_window',
+            'context': {
+                'is_modal': True, 'default_state':'draft',
```

Comment:
Why not doing a copy ?

Reply:
Because this is a regular model, and and if we cancel the modal, we should not create a record.

Reply:
Indeed... strange model. Ok, probably to cleanup in a near (?) future.